lib/ansible/module_utils/remote_management/intersight.py

```diff
@@ -0,0 +1,318 @@
+
+try:
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    HAS_REQUESTS = False
+
+intersight_argument_spec = dict(
+    api_private_key=dict(type='path', required=True),
+    api_uri=dict(type='str', default='https://intersight.com/api/v1'),
+    api_key_id=dict(type='str', required=True),
+    secure=dict(type='bool', default=True),
+    use_proxy=dict(type='bool', default=True),
+)
+
+
+def get_sha256_digest(data):
+    """
+    Generates a SHA256 digest from a String.
+
+    :param data: data string set by user
+    :return: instance of digest object
+    """
+
+    digest = hashlib.sha256()
+    digest.update(data.encode())
+
+    return digest
+
+
+def prepare_str_to_sign(req_tgt, hdrs):
+    """
+    Concatenates Intersight headers in preparation to be RSA signed
+
+    :param req_tgt : http method plus endpoint
+    :param hdrs: dict with header keys
+    :return: concatenated header authorization string
+    """
+    ss = ""
+    ss = ss + "(request-target): " + req_tgt.lower() + "\n"
+
+    length = len(hdrs.items())
+
+    i = 0
+    for key, value in hdrs.items():
+        ss = ss + key.lower() + ": " + value
+        if i < length - 1:
+            ss = ss + "\n"
+        i += 1
+
+    return ss
+
+
+def get_gmt_date():
+    """
+    Generated a GMT formatted Date
+
+    :return: current date
+    """
+
+    return formatdate(timeval=None, localtime=False, usegmt=True)
+
+
+class IntersightModule():
+
+    def __init__(self, module):
+        self.module = module
+        self.result = dict(changed=False)
+        if not HAS_CRYPTOGRAPHY:
+            self.module.fail_json(msg='cryptography is required for this module')
+        if not HAS_REQUESTS:
+            self.module.fail_json(msg='Requests is required for this module')
+        self.host = self.module.params['api_uri']
+        self.public_key = self.module.params['api_key_id']
+        self.private_key = open(self.module.params['api_private_key'], 'r').read()
+        self.secure = self.module.params['secure']
+        self.use_proxy = self.module.params['use_proxy']
+        self.digest_algorithm = 'rsa-sha256'
+        self.response_list = []
+
+    def get_rsasig_b64encode(self, data):
+        """
+        Generates an RSA Signed SHA256 digest from a String
+
+        :param digest: string to be signed & hashed
+        :return: instance of digest object
+        """
+
+        rsakey = serialization.load_pem_private_key(self.private_key.encode(), None, default_backend())
+        sign = rsakey.sign(data.encode(), padding.PKCS1v15(), hashes.SHA256())
+
+        return b64encode(sign)
+
+    def get_auth_header(self, hdrs, signed_msg):
+        """
+        Assmebled an Intersight formatted authorization header
+
+        :param hdrs : object with header keys
+        :param signed_msg: base64 encoded sha256 hashed body
+        :return: concatenated authorization header
+        """
+
+        auth_str = ""
+        auth_str = auth_str + "Signature"
```

Comment:
I don't really get this, why not just do.
```suggestion
        auth_str = "Signature"
```